app/services/create_github_repo_service.rb

```diff
@@ -0,0 +1,208 @@
+# frozen_string_literal: true
+
+# rubocop:disable ClassLength
+class CreateGitHubRepoService
+  attr_reader :entity, :stats_sender
+  delegate :assignment, :collaborator, :organization, :invite_status, to: :entity
+
+  def initialize(assignment, collaborator)
+    @entity = Entity.build(assignment, collaborator)
```

Comment:
Can you give `Entity` a more specific, descriptive name? I'm not sure what this class does.

Reply:
Even I'm not very happy with this name 😢 , do you have any suggestions? the job of this class is to manage `Assignment` and `GroupAssignment` related methods that have the same use, but different methods. e.g to get repos for `Assignment` would be `assignment.assignment_repos` and for `GroupAssignment` it would be `group_assignment.group_assignment_repos`. So with `Entity` we get a single method `repos`. And more such kind of methods.

Reply:
Possibly `Exercise` or `GenericAssignment`?

Reply:
I'm in favour of `Exercise`! Also, a few more renames like moving all related classes under the namespace `GitHubRepositoryCreator` and then rename `CreateGitHubRepositoryService1 -> `GitHubRepositoryCreator::Service` and rename `CreateGitHubRepositoryNewJob` -> `GitHubRepositoryCreator::Job`, what do you think?